client/src/js/components/bhDepotDropdown.js

```diff
@@ -22,9 +22,10 @@ function bhDepotController(Depot, AppCache, Notify) {
     .then(function (rows) {
       if (!rows.length) { return; }
       $ctrl.depots = rows;
-      $ctrl.selection = cache.selection || $ctrl.depots[0];
+      $ctrl.depotsAvailable = !!rows.length;
+      $ctrl.selection = cachedDepotExist() ? cache.selection : $ctrl.depots[0];
       $ctrl.loading = false;
```

Comment:
I know this is old code, but you do not need the `$ctrl.loading` line - it will be executed in the `.finally()` method.